include/mbedtls/md.h

```diff
@@ -236,61 +246,68 @@ const char *mbedtls_md_get_name( const mbedtls_md_info_t *md_info );
+ *                  with mbedtls_md_setup(), and before passing data with
+ *                  mbedtls_md_update().
  *
  * \param ctx       The generic message-digest context.
  *
- * \returns         \c 0 on success, #MBEDTLS_ERR_MD_BAD_INPUT_DATA if 
+ * \returns         \c 0 on success, #MBEDTLS_ERR_MD_BAD_INPUT_DATA if
  *                  parameter verification fails.
  */
 int mbedtls_md_starts( mbedtls_md_context_t *ctx );
 
 /**
- * \brief           This function feeds an input buffer into an ongoing 
+ * \brief           This function feeds an input buffer into an ongoing
  *                  message-digest computation.
  *
- *                  It is called between mbedtls_md_starts() and 
- *                  mbedtls_md_finish().
- *                  May be called repeatedly.
+ *                  You must call mbedtls_md_starts() before calling this
+ *                  function. You may call this function multiple times.
+ *                  Afterwards, call mbedtls_md_finish().
  *
  * \param ctx       The generic message-digest context.
  * \param input     The buffer holding the input data.
  * \param ilen      The length of the input data.
  *
- * \returns         \c 0 on success, #MBEDTLS_ERR_MD_BAD_INPUT_DATA if 
+ * \returns         \c 0 on success, #MBEDTLS_ERR_MD_BAD_INPUT_DATA if
  *                  parameter verification fails.
  */
 int mbedtls_md_update( mbedtls_md_context_t *ctx, const unsigned char *input, size_t ilen );
 
 /**
- * \brief           This function finishes the digest operation, 
+ * \brief           This function finishes the digest operation,
  *                  and writes the result to the output buffer.
  *
- *                  It is called after mbedtls_md_update(). It is usually 
- *                  followed by mbedtls_md_free() or mbedtls_md_starts().
+ *                  Call this function after a call to mbedtls_md_starts(),
+ *                  followed by any number of calls to mbedtls_md_update().
+ *                  Afterwards, you may either clear the context with
+ *                  mbedtls_md_free(), or call mbedtls_md_starts() to reuse
+ *                  the context for another digest operation with the same
+ *                  algorithm.
  *
  * \param ctx       The generic message-digest context.
  * \param output    The buffer for the generic message-digest checksum result.
  *
- * \returns         \c 0 on success, or #MBEDTLS_ERR_MD_BAD_INPUT_DATA if 
+ * \returns         \c 0 on success, or #MBEDTLS_ERR_MD_BAD_INPUT_DATA if
  *                  parameter verification fails.
  */
 int mbedtls_md_finish( mbedtls_md_context_t *ctx, unsigned char *output );
 
 /**
  * \brief          This function calculates the message-digest of a buffer,
- *                 with respect to a configurable message-digest algorithm.
- *                 The result is calculated as 
+ *                 with respect to a configurable message-digest algorithm
+ *                 in a single cell.
```

Comment:
Blocker: Typo `cell` -> `call`. @gilles-peskine-arm Were you ok with the wording?

Reply:
fixed

Reply:
I don't like “configurable”, because it makes me think that there are separate calls to set a configuration and to use that configuration. But that's minor.